src/main/java/org/montrealjug/billetterie/ui/RegistrationController.java

```diff
@@ -207,47 +241,40 @@ public String verifyBooker(@PathVariable String signature, Model model) {
 
             // Return booker-activities template
             return "booker-activities";
-        } else if (optionalEvent.isPresent()) {
-            // Booker not found but event exists, return index with error
-            Event event = optionalEvent.get();
-            PresentationEvent presentationEvent =
-                    new PresentationEvent(
-                            event.getId(),
-                            event.getTitle(),
-                            event.getDescription(),
-                            event.getDate(),
-                            toIndexActivities(event.getActivities()),
-                            event.isActive(),
-                            event.getImagePath());
-
-            model.addAttribute("event", presentationEvent);
-            model.addAttribute("error", "The booker could not be retrieved from the DB");
-
-            return "index";
         } else {
-            // No active event, return index
+            // or the Booker is missing, or there is no active Event, so we go to `index`
+            if (event != null) {
+                // add the Event for index, if any
+                PresentationEvent presentationEvent =
+                        new PresentationEvent(
+                                event.getId(),
+                                event.getTitle(),
+                                event.getDescription(),
+                                event.getDate(),
+                                toIndexActivities(event.getActivities()),
+                                event.isActive(),
+                                event.getImagePath());
+                model.addAttribute("event", presentationEvent);
+            }
+            if (booker == null) {
+                // set the error msg if no Booker found
+                model.addAttribute("error", "The booker could not be retrieved from the DB");
+            }
             return "index";
         }
     }
 
-    String retrieveBaseUrl(HttpServletRequest request) {
+    static String retrieveBaseUrl(HttpServletRequest request) {
         String requestURL = request.getRequestURL().toString();
         String requestURI = request.getRequestURI();
 
         return requestURL.substring(0, requestURL.length() - requestURI.length())
                 + request.getContextPath();
     }
 
-    boolean isSameParticipant(Participant p, @Valid ParticipantSubmission participantSub) {
+    static boolean isSameParticipant(Participant p, ParticipantSubmission participantSub) {
```

Comment:
I'm curious: why did you drop the `@Valid` ?

Reply:
wow! thanks for the lengthy explanation!
OK, already validated in the controller and yes, does not work in a static context!
okee dok! thanks!